main.js

```diff
@@ -24,21 +17,101 @@ function renderItem(product) {
 
   $newItem.classList.add('item-container')
   $newItem.classList.add('col-md-10')
+  $newItem.setAttribute('data-product', product.id)
+  $newItem.id = product.id
+
   $itemImg.classList.add('item-img')
   $itemImg.setAttribute('src', product.img)
+  $itemImg.setAttribute('data-product', product.id)
   $itemImg.classList.add('col-md-4')
+
   $itemName.classList.add('item-name')
   $itemName.classList.add('col-md-4')
+  $itemName.setAttribute('data-product', product.id)
   $itemName.textContent = product.name
+
   $itemPrice.classList.add('item-stats')
   $itemPrice.classList.add('col-md-4')
+  $itemPrice.setAttribute('data-product', product.id)
   $itemPrice.textContent = '$' + product.price.toFixed(2)
+
   $lineBreak.classList.add('col-md-12')
+
   $newItem.appendChild($itemImg)
   $newItem.appendChild($itemName)
   $newItem.appendChild($itemPrice)
   $newItem.appendChild($lineBreak)
 
   return $newItem
 }
+
+function displayItems(itemList) {
+  var $holder = document.getElementById('item-holder')
+  for (var i = 0; i < itemList.length; i++) {
+    var currentProduct = itemList[i]
+    var $product = renderItem(currentProduct)
+    $holder.appendChild($product)
+  }
+  var $itemContainer = document.querySelectorAll('div.item-container')
+  $holder.addEventListener('click', function(event) {
+    var targetData = event.target.dataset.product
+    console.log(typeof targetData + 'w')
+    console.log(targetData + 'x')
+    selectedItem(itemList, targetData, $itemContainer, $holder)
+  })
+}
+
 displayItems(items)
+
+// Issue-2
+
+function selectedItem(itemList, target, container, row) {
+  var view = 'selected item page'
+  var $details = document.getElementById('item-details')
+  $details.innerHTML=''
+  console.log(container[target].id)
+  var chosenItemId = container[target].id
+  var product = renderItem(items[chosenItemId])
+  $details.appendChild(product)
+  views(view, row, $details)
+  itemDescription(itemList, target, product)
+  createReturnButton(product, row, $details)
```

Comment:
in this function, instead of relying on the DOM position, just rely on the `productId` that we're interested in. Write another function that will take that id, and return to you the object in the array that you want.

```js
function findProduct(productId, products) {
  // search through products until you find the one with productId 
}
```